torch/testing/_asserts.py

```diff
@@ -157,7 +228,13 @@ def _check_attributes_equal(
     if check_dtype and actual.dtype != expected.dtype:
         return AssertionError(msg_fmtstr.format("dtype", actual.dtype, expected.dtype))
 
-    if check_stride and actual.stride() != expected.stride():
+    if actual.is_sparse != expected.is_sparse:
```

Comment:
This block needs a comment

Reply:
Let me know if we still need an elaborate comment after the redesign of the checks.